lib/urllib_kerberos/__init__.py

```diff
@@ -94,35 +88,37 @@ def generate_request_header(self, req, headers, neg_value):
             kwargs['principal'] = user
             kwargs['password'] = password
 
-        result, self.context = k.authGSSClientInit("HTTP@%s" % domain,
-                                                   **kwargs)
+        kwargs['mech_oid'] = self.mech_oid
+
+        result, req.kerberos_context = k.authGSSClientInit("HTTP@%s" % domain,
```

Comment:
So we're going to put the context on the request instead of on the current object? Is there any precedent for doing that in other authentication handlers?